python/run-tests.py

```diff
@@ -97,7 +97,12 @@ def run_cmd(cmd, throw_on_error=True, env=None, stream_output=False, **kwargs):
         return (exit_code, stdout, stderr)
 
 
+def run_python_style_checks():
+    run_cmd([os.path.join(SPARK_HOME, "dev", "lint-python")])
+
+
 if __name__ == "__main__":
     root_dir = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
     package = prepare(root_dir)
     test(root_dir, package)
+    run_python_style_checks()
```

Comment:
I think it would benefit having the style checks before running tests as it would be annoying spending half an hour running tests and to find out that a space is missing somewhere and to have to re-run the tests. can we have it as a part of the root `run-tests.py`

Reply:
Yep. I thought of the same. This is my first pass though so I don't even know if the script works yet.

Reply:
@rahulsmahadev it is probably ready for a more open review now. I am going to bring the PR out of draft